Payload_Type/apollo/agent_code/Apollo/Utils/DarkMelkor.cs

```diff
@@ -0,0 +1,246 @@
+using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Apollo.Utils
+{
+    class DarkMelkor
+    {
+        [DllImport("kernel32")]
+        static extern bool VirtualProtect(IntPtr lpAddress, UIntPtr dwSize, uint flNewProtect, out uint lpflOldProtect);
+
+        // API
+        //======================
+        [DllImport("ntdll.dll")]
+        public static extern UInt32 NtFreeVirtualMemory(
+            IntPtr ProcessHandle,
+            ref IntPtr BaseAddress,
+            ref IntPtr RegionSize,
+            AllocationType FreeType);
+
+        [DllImport("ntdll.dll")]
+        public static extern void RtlZeroMemory(
+            IntPtr Destination,
+            int length);
+
+        [DllImport("kernel32.dll")]
+        public static extern IntPtr LocalFree(
+            IntPtr hMem);
+
+        [DllImport("crypt32.dll", CharSet = CharSet.Auto)]
+        public static extern bool CryptProtectData(
+            ref DATA_BLOB pPlainText,
+            string szDescription,
+            ref DATA_BLOB pEntropy,
+            IntPtr pReserved,
+            IntPtr pPrompt,
+            int dwFlags,
+            ref DATA_BLOB pCipherText);
+
+        [DllImport("crypt32.dll", CharSet = CharSet.Auto)]
+        public static extern bool CryptUnprotectData(
+            ref DATA_BLOB pCipherText,
+            ref string pszDescription,
+            ref DATA_BLOB pEntropy,
+            IntPtr pReserved,
+            IntPtr pPrompt,
+            int dwFlags,
+            ref DATA_BLOB pPlainText);
+
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+        internal struct DATA_BLOB
+        {
+            public int cbData;
+            public IntPtr pbData;
+        }
+
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+        internal struct CRYPTPROTECT_PROMPTSTRUCT
+        {
+            public int cbSize;
+            public int dwPromptFlags;
+            public IntPtr hwndApp;
+            public string szPrompt;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct DPAPI_MODULE
+        {
+            public String sModName;
+            public int iModVersion;
+            public int iModSize;
+            public IntPtr pMod;
+            public Byte[] bMod;
+        }
+
+        [Flags]
+        public enum AllocationType : uint
+        {
+            Commit = 0x1000,
+            Reserve = 0x2000,
+            Decommit = 0x4000,
+            Release = 0x8000,
+            Reset = 0x80000,
```

Comment:
All these definitions should exist in the correct file under the Native folde/namespace. There should be things for methods, enums, constants, etc.

Reply:
Moved the correspondent P/Invoke and Enums to the correct files.